chain/consensus/mir/manager.go

```diff
@@ -270,18 +266,24 @@ func (m *Manager) Serve(ctx context.Context) error {
 
 		select {
 		case <-ctx.Done():
-			log.With("validator", m.id).Debug("Mir manager: context closed")
-			return nil
+			log.With("validator", m.id).Info("Mir manager: context closed")
 
-		// First catch potential errors when mining.
-		case err := <-mirErrChan:
-			log.With("validator", m.id).Info("manager received error:", err)
+			// Perform cleanup of Node's modules and ensure that mir is closed when we stop mining.
+			m.Stop()
```

Comment:
It is quite un-intuitive to me to use an exported `Stop` function this way.
Especially since the code actually relies on this being the only function from which `Stop` is called. If it were called from anywhere else, the code panics (on line 285).

However, if this indeed is the only function from where we call `Stop` (as it currently must be - if it's not, we have a problem), there is no graceful way of stopping the Manager except for canceling its context. Maybe we don't need it, in which case it's totally fine, but it should be a conscious choice. In this case I strongly suggest to at least make it not exported.

Reply:
`stop` is unexported now.